### DISCUSSION TITLE
How to save more than 3 checkpoints.

### DISCUSSION THREAD

**Discussion by dtemnov** [NVIDIA-NeMo/NeMo#3841] [Q&A] [answered]
Hi,

It should be very easy code/config change but I couldn't find the proper way to pass the top-k param from PTL(I see that Nemo is extending PTL ModelCheckpoint class a litle bit). So the question is - How to save more than 3 best checkpoints(and the last one)? I would like to validate my FastPitch model during finetuning because val_loss doesn't usually make sense in this case.

Thanks a lot for your help folks!

**Comment by VahidooX** [comment, marked as answer]
Hi, you may add save_top_k to your exp_manager like here: https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/conformer/conformer_transducer_bpe.yaml#L238

Also you may set always_save_nemo=true so that it stores the checkpoints in nemo format too so that you can evaluate them easily.

**Reply by dtemnov** [reply]
Thanks for your help. That solves my issue.
